argocd/datadog_checks/argocd/check.py

```diff
@@ -0,0 +1,105 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from six import PY2
+
+from datadog_checks.base import ConfigurationError, OpenMetricsBaseCheckV2
+from datadog_checks.base.constants import ServiceCheck
+
+from .config_models import ConfigMixin
+from .metrics import API_SERVER_METRICS, APPLICATION_CONTROLLER_METRICS, REPO_SERVER_METRICS
+
+API_SERVER_NAMESPACE, APP_CONTROLLER_NAMESPACE, REPO_SERVER_NAMESPACE = [
+    'argocd.api_server',
+    'argocd.app_controller',
+    'argocd.repo_server',
+]
+
+
+class ArgocdCheck(OpenMetricsBaseCheckV2, ConfigMixin):
+    DEFAULT_METRIC_LIMIT = 0
+
+    def __init__(self, name, init_config, instances):
+        if PY2:
+            raise ConfigurationError(
```

Comment:
I wonder if we can define this checking for py2 in one central location (base pkg) and then just apply it more declaratively to relevant integrations...

Reply:
good thing to think about but way out of scope here

Reply:
Yup, not a blocker here at all!